src/PoolManager.sol

```diff
@@ -184,16 +184,20 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         IPoolManager.ModifyPositionParams memory params,
         bytes calldata hookData
     ) external override noDelegateCall onlyByLocker returns (BalanceDelta delta) {
+        PoolId id = key.toId();
```

Comment:
Why move this to the top level? Seems cleaner to have it in` Pool.sol`

Reply:
Because if its in the pool, then its _after_ the hook call. So that means on an uninitialized pool, a `swap` transaction succeeds with a hook that NoOps (even though the pool doesnt exist). So I've moved it top level so that we can maintain the invariant that actions on an uninitialized pool always revert

Reply:
maybe new modifier `onlyInitializedPool(key)`? that duplicates id hashing in code but id guess it gets compiled out, could check?

Reply:
modifier or helper func sgtm :)

Reply:
Modifier put gas up by 1000 for swaps because of repeated hashing. I've put in a helper function instead